micropip/_utils.py

```diff
@@ -88,7 +89,11 @@ def parse_wheel_filename(
 
 # TODO: Move these helper functions back to WheelInfo
 def parse_version(filename: str) -> Version:
-    return parse_wheel_filename(filename)[1]
+    """Parses a version from a wheel filename, returning a CachedVersion."""
+    version_obj = parse_wheel_filename(filename)[1]
+    if isinstance(version_obj, CachedVersion):
+        return version_obj
+    return CachedVersion(str(version_obj))
```

Comment:
parse_wheel_filename will always return `Version` object. Changing it to str and converting it to `CachedVersion` again is wasteful. How about creating a helper function, something like

```python
CachedVersion.from_version(version_obj)
```